src/nplinker/scoring/link_graph.py

```diff
@@ -192,5 +204,67 @@ def get_link_data(
         Returns:
             A dictionary of scoring methods and their data for the link between the two objects, or
             None if there is no link between the two objects.
+
+        Examples:
+            >>> lg.get_link_data(gcf, spectrum)
+            {"metcalf": Score("metcalf", 1.0, {"cutoff": 0.5})}
         """
         return self._g.get_edge_data(u, v)  # type: ignore
+
+    def filter(self, u_nodes: Sequence[Entity], v_nodes: Sequence[Entity] = [], /) -> LinkGraph:
+        """Return a new LinkGraph object with the filtered links between the given objects.
+
+        The new LinkGraph object will only contain the links between `u_nodes` and `v_nodes`.
+
+        If `u_nodes` or `v_nodes` is empty, the new LinkGraph object will contain the links for
+        the given objects in `v_nodes` or `u_nodes`, respectively. If both are empty, return an
+        empty LinkGraph object.
+
+        Note that not all objects in `u_nodes` and `v_nodes` need to be present in the original
```

Comment:
About this note, when using only one sequence of objects (`u_nodes` or `v_nodes`) it throws a `KeyError` because of the `__getitem__` method called during the `_filter_one_node` method. It works without errors instead when using two sequences of objects (`u_nodes` and `v_nodes`), calling `_filter_two_nodes`.

Reply:
Good catch! it's a bug. Fixed in the new commit.